libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -905,12 +915,47 @@ internal class MapRouteLine(
         }
     }
 
+    /**
+     * Updates the route line appearance from the origin point to the point indicated in
+     * @param point representing the portion of the route that has been traveled.
+     */
+    fun updateTraveledRouteLine(point: Point) {
+        ifNonNull(primaryRoute, primaryRoute?.distance()) { primaryRoute, routeDistance ->
+            val lineString = getLineStringForRoute(primaryRoute)
+            val nearestPointOnLineDistance = TurfMisc.nearestPointOnLine(
+                point,
+                lineString.coordinates(),
+                TurfConstants.UNIT_METERS
+            ).getProperty("dist").asDouble
+
+            if (
+                nearestPointOnLineDistance > ROUTE_LINE_UPDATE_MAX_DISTANCE_THRESHOLD_IN_METERS ||
+                vanishingPointUpdateInhibited
```

Comment:
Check for the `vanishingPointUpdateInhibited` can be executed before we do the turf calculations.